File: services/payment-service/controllers/paymentController.js
require('dotenv').config();
const Stripe = require('stripe');
const stripe = Stripe(process.env.STRIPE_SECRET_KEY); // Secret Key from Stripe Dashboard

exports.handlePayment = async (req, res) => {
  console.log("received payment request")
  const { amount } = req.body; // Amount in cents

  const customer = await stripe.customers.create();
  const ephemeralKey = await stripe.ephemeralKeys.create(
    {customer: customer.id},
    {apiVersion: '2024-11-20.acacia'}
  );
  
  try {
    const paymentIntent = await stripe.paymentIntents.create({
      amount: amount,
      currency: 'usd',
      customer: customer.id,
      // In the latest version of the API, specifying the `automatic_payment_methods` parameter
      // is optional because Stripe enables its functionality by default.
      automatic_payment_methods: {
        enabled: true,
      },
    });
  
    res.json({
      paymentIntent: paymentIntent.client_secret,
      ephemeralKey: ephemeralKey.secret,
      customer: customer.id,
      publishableKey : process.env.STRIPE_PUBLISHABLE_KEY
    });
  } catch (error) {
    console.error("Error creating payment intent:", error);
    res.status(500).send({ error: error.message });
  }
};

File: services/payment-service/controllers/paymentController.py
from flask import Flask, request, jsonify
import os
import stripe
from dotenv import load_dotenv

# Load environment variables from a .env file
load_dotenv()

# Initialize Stripe with your secret key
stripe.api_key = os.getenv('STRIPE_SECRET_KEY')

app = Flask(__name__)

@app.route('/handlePayment', methods=['POST'])
async def handle_payment():
    print("Received payment request")
    data = request.json
    amount = data.get('amount')  # Amount in cents

    try:
        # Create a new customer
        customer = stripe.Customer.create()

        # Create an ephemeral key for the customer
        ephemeral_key = stripe.EphemeralKey.create(
            {"customer": customer.id},
            api_version='2024-11-20.acacia'
        )

        # Create a payment intent
        payment_intent = stripe.PaymentIntent.create(
            amount=amount,
            currency='usd',
            customer=customer.id,
            automatic_payment_methods={
                'enabled': True,
            },
        )

        return jsonify({
            'paymentIntent': payment_intent.client_secret,
            'ephemeralKey': ephemeral_key.secret,
            'customer': customer.id,
            'publishableKey': os.getenv('STRIPE_PUBLISHABLE_KEY')
        })

    except Exception as error:
        print("Error creating payment intent:", error)
        return jsonify({'error': str(error)}), 500

if __name__ == '__main__':
    app.run(debug=True)
